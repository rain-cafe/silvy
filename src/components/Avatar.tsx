import styles from './Avatar.module.scss';
import Image from 'next/image';

type AvatarProps = {
    src: string;
}

export default function Avatar({
    src
}: AvatarProps) {
    return (
        <Image
            className={styles.avatar}
            width={40}
            height={40}
            src={src}
        />
    )
}