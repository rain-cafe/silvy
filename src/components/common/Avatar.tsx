import styles from './Avatar.module.scss';
import Image from 'next/image';

export type AvatarProps = {
    src: string;
    size?: number;
}

export function Avatar({
    src,
    size = 40
}: AvatarProps): JSX.Element {
    return (
        <Image
            alt='Avatar'
            width={size}
            height={size}
            className={styles.avatar}
            src={src}
            priority
        />
    )
}