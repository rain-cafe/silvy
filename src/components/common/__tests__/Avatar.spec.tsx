import { createRenderComponent } from '@rain/test-utils/render';
import { Avatar, AvatarProps } from '../Avatar'

describe('Component(Avatar)', () => {
    const renderAvatar = createRenderComponent<AvatarProps>(Avatar, {
        src: 'https://silvy.rains.cafe'
    });

    it('should display the Avatar', () => {
        const { baseElement } = renderAvatar()

        expect(baseElement).toMatchSnapshot()
    });
});
