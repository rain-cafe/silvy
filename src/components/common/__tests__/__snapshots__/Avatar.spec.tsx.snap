// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component(Avatar) should display the Avatar 1`] = `
<body>
  <div>
    <img
      alt="Avatar"
      class="avatar"
      data-nimg="1"
      decoding="async"
      height="40"
      src="/_next/image?url=https%3A%2F%2Fsilvy.rains.cafe&w=96&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fsilvy.rains.cafe&w=48&q=75 1x, /_next/image?url=https%3A%2F%2Fsilvy.rains.cafe&w=96&q=75 2x"
      style="color: transparent;"
      width="40"
    />
  </div>
</body>
`;
